src/core_modules/capture-core/rules/actionsCreator/postProcessRulesEffects.js

```diff
@@ -116,12 +116,15 @@ function filterFieldsHideEffects(
     }
 
     const elementIds = Object.keys(hideEffects);
+    const compulsoryElements = foundation.getElements().filter(({ compulsory }) => compulsory);
+
     return elementIds
         .filter((elementId) => {
-            const element = foundation.getElement(elementId);
-            const dataCompulsory = element && element.compulsory;
+            const { compulsory: elementIsCompulsory } =
+                compulsoryElements.find(({ id }) => id === elementId) || {};
+
             const compulsoryEffect = makeCompulsoryEffects[elementId];
-            return !(dataCompulsory || compulsoryEffect);
+            return !(elementIsCompulsory || compulsoryEffect);
```

Comment:
What we did here is look in _all_ the elements and get back the id with the `elementId`.  This specific action costs us time and made the application slower. 

Also is important to understand we care only about the `compulsory` elements. 

What I do now is filter the elements and keep only the ones that are `compulsory` which are usually few and the `find` in these the one with the `elementId` we are looking for.